### ISSUE TITLE
Grasshopper_Toolkit: Casting issue after BHoM geometry output

### ISSUE THREAD

**Issue by pawelbaran** [BHoM/Grasshopper_Toolkit#539]
#### Description:

Once a BHoM geometry is output in GH, it looks as if was a native GH/Rhino type (Trimmed Surface below), but it often cannot be processed using native GH methods as long as not cast back:

![image](https://user-images.githubusercontent.com/26874773/93458860-2cbaef80-f8e1-11ea-845f-22ca985e8f60.png)

It is a quite annoying issue, particularly to the non-expert users.

#### Test file(s):

On [SharePoint](https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FTest%20Scripts%2FGrasshopper%5FToolkit%2FGrasshopper%5FToolkit%2DIssue539%2DCastGeometry&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4)

**Comment by adecler** [comment]
Ok, I haven't touched that part of the code in a very long time so I am a bit rusty in the specifics of GH. @epignatelli , feel free to jump in if I am missing the obvious. In the meantime, I will document what  I find here.

Any Grasshopper Param (including the BHoM parameters we created) has two casting methods:
- `CastFrom(object source)` :  this is called when the parameter is reading its input value (i.e. when the component that own the parameter is being run).
- `CastTo<Q>(ref Q target)` this is called when the parameter is being asked to provide it value to another parameter (i.e. when the param is plugged into the input of another component and that component is executed).

This is a great feature from GH because it means we don't have to constantly convert back and forth between BHoM and Grasshopper when we are chaining BHoM components (we can just stay in the BHoM realm). So big improvement in efficiency an reduction of risk of errors due to casting (e.g. nurbs curves)

Here's  where the problem lies in this case: for some reason the GH Area component is not asking the PlanarSurface output param to cast its content. As we can see, it works fine for things like the GH component that calculate the distance between points.

![image](https://user-images.githubusercontent.com/16853390/93553300-3c037100-f9a5-11ea-97a3-3e0e8a80a61f.png)

Since the input of the `Area` component is a `Geometry` param, I have tried this:

![image](https://user-images.githubusercontent.com/16853390/93553470-a3b9bc00-f9a5-11ea-8ed6-72a1ae04a54e.png)

And, to no big surprise, the `Geometry` param is the culprit as it does not casting 😢 

This is where I am a bit unsure what to do and you might have some idea @epignatelli . We obviously want to keep the feature that BHoM geometries are only casted when needed and we cannot modify the geometry param ourselves. Detecting that the next param of our output is a GH geometry is no good either since we can have multiple outputs.
